files/class-vip-filesystem-stream-wrapper.php

```diff
@@ -316,17 +316,22 @@ public function stream_flush() {
 
 		try {
 			// Upload to file service
-			$result                   = $this->client->upload_file( $this->uri, $this->path );
-			$this->should_flush_empty = false;
+			$result = $this->client->upload_file( $this->uri, $this->path );
 			if ( is_wp_error( $result ) ) {
 				trigger_error(
 					sprintf( 'stream_flush failed for %s with error: %s #vip-go-streams', esc_html( $this->path ), esc_html( $result->get_error_message() ) ),
 					E_USER_WARNING
 				);
 
+				if ( $this->should_flush_empty ) {
+					$this->should_flush_empty = false;
+					$this->client->cache_file_stats( $this->path, [] );
```

Comment:
There is no `clear_file_stats()` exposed by the API client, but passing an empty array effectively discards the cache.

Reply:
Can we add a note in the code for why we are passing an empty array?

Reply:
Done in aa75268